lib.php

```diff
@@ -103,3 +103,44 @@ function cpdlogbook_extend_settings_navigation($settings, $cpdlogbooknode) {
         new moodle_url('/mod/cpdlogbook/periods.php', ['id' => $PAGE->cm->id])
     );
 }
+
+/**
+ * This function is a required callback function, currently missing some security checks I think
+ * From the File_API documentation (https://docs.moodle.org/dev/File_API):
+ *
+ * Serve the files from the MYPLUGIN file areas
+ *
+ * @param stdClass $course the course object
+ * @param stdClass $cm the course module object
+ * @param stdClass $context the context
+ * @param string $filearea the name of the file area
+ * @param array $args extra arguments (itemid, path)
+ * @param bool $forcedownload whether or not force download
+ * @param array $options additional options affecting the file serving
+ * @return bool false if the file not found, just send the file otherwise and do not return anything
+ */
+function mod_cpdlogbook_pluginfile($course, $cm, $context, $filearea, $args, $forcedownload, array $options=array()) {
+
+    if ($filearea != 'attachments') {
```

Comment:
This is missing some access control checks, which should be broadly exactly the same as if you can view the page where you can click the download button

Reply:
Could you check if it's all there now?

Reply:
It checks that you are logged in and can view the cpd instance, but I think it is missing a check for if you are the owner of this particular item. It should be consistent with the checks here:

https://github.com/brendanheywood/moodle-mod_cpdlogbook/blob/main/details.php#L40-L49

Ideally this should be refactored into the same code called from both places